File: src/main/frontend/src/app/services/authentication.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import {
  AuthenticationRequest,
  UserDetails,
} from '../models/authentication.models';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class AuthenticationService {
  private httpClient = inject(HttpClient);
  private baseUrl = 'http://localhost:8080/auth';

  public authenticate(authRequest: AuthenticationRequest) {
    return this.httpClient.post<UserDetails>(
      `${this.baseUrl}/login`,
      authRequest,
      {
        withCredentials: true,
      }
    );
  }

  public register(authRequest: AuthenticationRequest) {
    return this.httpClient.post<UserDetails>(
      `${this.baseUrl}/register`,
      authRequest
    );
  }

  public getCurrentLoggedUser() {
    return this.httpClient.get<UserDetails>(`${this.baseUrl}/current-user`);
  }
}

File: src/main/java/pu/fmi/slavnarech/services/security/AuthenticationService.java
package pu.fmi.slavnarech.services.security;

import org.springframework.security.core.userdetails.UserDetails;
import pu.fmi.slavnarech.entities.user.security.AuthenticationRequest;
import pu.fmi.slavnarech.entities.user.security.AuthenticationResponse;

public interface AuthenticationService {

  AuthenticationResponse register(AuthenticationRequest authRequest);

  AuthenticationResponse authenticate(AuthenticationRequest authRequest);

  AuthenticationResponse getRefreshToken(String refreshToken);

  UserDetails getCurrentLoggedInUser();
}
